common/src/main/java/com/wynntils/services/mapdata/providers/builtin/WaypointsProvider.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.mapdata.providers.builtin;
+
+import com.wynntils.services.map.pois.CustomPoi;
+import com.wynntils.services.mapdata.attributes.AbstractMapAttributes;
+import com.wynntils.services.mapdata.attributes.impl.AlwaysMapVisibility;
+import com.wynntils.services.mapdata.attributes.impl.FadingMapVisiblity;
+import com.wynntils.services.mapdata.attributes.impl.NeverMapVisibility;
+import com.wynntils.services.mapdata.attributes.type.MapAttributes;
+import com.wynntils.services.mapdata.attributes.type.MapVisibility;
+import com.wynntils.services.mapdata.type.MapFeature;
+import com.wynntils.services.mapdata.type.MapLocation;
+import com.wynntils.utils.mc.type.Location;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class WaypointsProvider extends BuiltInProvider {
+    private static final List<MapFeature> PROVIDED_FEATURES = new ArrayList<>();
+    private static int counter;
+
+    @Override
+    public String getProviderId() {
+        return "waypoints";
+    }
+
+    @Override
+    public Stream<MapFeature> getFeatures() {
+        return PROVIDED_FEATURES.stream();
+    }
+
+    public static void resetFeatures() {
+        PROVIDED_FEATURES.clear();
+    }
+
+    public static void registerFeature(CustomPoi customPoi) {
+        int tier =
+                switch (customPoi.getIcon()) {
+                    case CHEST_T1 -> 1;
+                    case CHEST_T2 -> 2;
+                    case CHEST_T3 -> 3;
+                    case CHEST_T4 -> 4;
+                    default -> 0;
+                };
+        if (tier == 0) {
+            String iconId = MapIconsProvider.getIconIdFromTexture(customPoi.getIcon());
+            PROVIDED_FEATURES.add(new WaypointLocation(
+                    customPoi.getLocation().asLocation(), customPoi.getName(), iconId, customPoi.getVisibility()));
+        } else {
+            PROVIDED_FEATURES.add(new FoundChestLocation(customPoi.getLocation().asLocation(), tier));
+        }
```

Comment:
This really wants us to not collect loot chest locations to the personal POI list, right? Most likely we should move all loot chest locations to a new list of storage, and separate it here.

Reply:
Yes, I think that was a huge mistake from out part (copied over from Legacy). The stuff we collect automatically should be kept separate from manually added waypoints. 

For instance, automatically collected chests might be checked over time if they still exist or needs to be removed, etc. With the new looting system, I guess it would make sense to store the timestamp of when such a chest was last visited, etc.

I was thinking about just ditching all these chests completely, but my assumption here is that anything with a tier X chest icon means it is automatically added (not necessarily correct) and that we should take ownership of it. I guess a more proper way to do this would be to run some kind of conversion that splits out chests from the rest of the waypoints, and ask the user what to do with them.

Reply:
Do you think we should do the separation now? We can just leave all old chests in the user waypoints, and collect them into a new list now. I know, it's less than ideal, but I don't think we want to deal with anything else. The users can delete these chests from their custom waypoints in even the old waypoint manager. We will render the newly loot chests, but they won't be editable for a short period.

Reply:
Yes, we should start doing a separate collection as soon as possible. Then we could correlate the two sets and delete the "user waypoints" that match coordinate and icon type. I think that would be the best. But the sooner we start collecting, the better.

Otoh, I would also like to have the collected data in the new mapdata json format, so until that is finished, I think we should hold off collecting. Or, perhaps, we can store it as a HiddenConfig or Storage for the time being, and then convert it?

Reply:
I am not sure how to make it be in the new format. But we can't upfix storage, so perhaps let's not do it atm.

Reply:
Yeah, we can discuss that further on discord.